File: scripts/python/model_training.py
import pandas as pd
from sklearn.linear_model import LogisticRegression
from sklearn.model_selection import train_test_split
from sklearn.metrics import roc_auc_score

df = pd.read_csv("outputs/synthetic_compliance_data.csv")

X = df[["adverse_event_reports", "inspection_findings", "previous_violations", "compliance_training"]]
y = df["compliance_issue"]

X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.3, random_state=42)

model = LogisticRegression()
model.fit(X_train, y_train)

y_pred = model.predict_proba(X_test)[:, 1]
auc = roc_auc_score(y_test, y_pred)

X_test["predicted"] = y_pred
X_test["actual"] = y_test.values
X_test.to_csv("outputs/predictions.csv", index=False)

params = pd.DataFrame({
    "feature": X.columns,
    "coefficient": model.coef_[0]
})
params.to_csv("outputs/model_coefficients.csv", index=False)

with open("outputs/model_metrics.csv", "w") as f:
    f.write(f"ROC_AUC,{auc:.4f}")

File: scripts/r/model_training.R
rm(list = ls())

df <- read.csv("data/synthetic_compliance_data.csv")

# Select relevant variables
X <- df[, c("adverse_event_reports", "inspection_findings", "previous_violations", "compliance_training")]
y <- df$compliance_issue

# Train-test split
set.seed(42)
split <- createDataPartition(y, p = 0.7, list = FALSE)
train <- df[split, ]
test <- df[-split, ]

model <- glm(compliance_issue ~ adverse_event_reports + inspection_findings +
               previous_violations + compliance_training,
             data = train, family = binomial)

test$predicted <- predict(model, newdata = test, type = "response")
test$prediction_class <- ifelse(test$predicted > 0.5, 1, 0)

write.csv(test, "outputs/predictions.csv", row.names = FALSE)

# Write metrics
auc_val <- roc(test$compliance_issue, test$predicted)$auc
metrics <- paste("ROC_AUC,", round(auc_val, 4))
write(metrics, "outputs/model_metrics.csv")
